include/obstack.h

```diff
@@ -0,0 +1,459 @@
+ *
+ ****************************************************************************/
+#define obstack_next_free(h) ((h)->next_free)
+
+/****************************************************************************
+ * Name: obstack_blank_fast
+ *
+ * Description:
+ *   Moves the end of the currently growing object by given size and thus
+ *   adding given number of uninitialized bytes to the growing object.
+ *   There is no check if there is enough room and thus it is easy to cause
+ *   buffer overrun. Use only when you are sure that there is enough room!
+ *
+ * Input Parameters:
+ *   h: pointer to the handle used to grow the object.
+ *   size: number of bytes
+ *
+ * Returned Value:
+ *   The new address just after the end of the currently growing object.
+ *
+ ****************************************************************************/
+#define obstack_blank_fast(h, size) ((h)->next_free += size)
+
+/****************************************************************************
+ * Name: obstack_1grow_fast
+ *
+ * Description:
+ *   Adds one byte to the currently growing object.
+ *   There is no check if there is enough room and thus it is easy to cause
+ *   buffer overrun. Use only when you are sure that there is enough room!
+ *
+ * Input Parameters:
+ *   h: pointer to the handle used to grow the object.
+ *   data: byte to be added
+ *
+ * Returned Value:
+ *   Added byte.
+ *
+ ****************************************************************************/
+#define obstack_1grow_fast(h, data) (*((h)->next_free++) = data)
+
+/****************************************************************************
+ * Public Type Definitions
+ ****************************************************************************/
+
+struct _obstack_chunk           /* Chunk head. */
+{
+  char *limit;                  /* address of char after this chunk */
+  struct _obstack_chunk *prev;  /* address of prior chunk or NULL */
+};
+
+struct obstack
+{
+  size_t chunk_size;            /* preferred size to allocate chunks in */
+  struct _obstack_chunk *chunk; /* address of current struct _obstack_chunk */
+  char *object_base;            /* address of object we are building */
+  char *next_free;              /* where to add next char to current object */
+};
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+#if defined(__cplusplus)
+extern "C"
+{
+#endif
+
+/****************************************************************************
+ * Name: obstack_init
+ *
+ * Description:
+ *   Initialize obstack for allocation of objects.
+ *   Compared to the GlibC version this won't initialize a first chunk.
+ *
+ * Input Parameters:
+ *   h: pointer to the handle to initialize
+ *
+ ****************************************************************************/
+
+void obstack_init (FAR struct obstack *h);
```

Comment:
```suggestion
void obstack_init(FAR struct obstack *h);
```
remove all space before (

Reply:
Ok